test/vstest.console.UnitTests/Processors/CLIRunSettingsArgumentProcessorTests.cs

```diff
@@ -258,6 +272,78 @@ public void InitializeShouldNotUpdateCommandLineOptionsArchitectureAndFxIfNotPro
             Assert.IsFalse(this.commandLineOptions.FrameworkVersionSpecified);
         }
 
+        [DynamicData(nameof(TestRunParameterArgValidTestCases), DynamicDataSourceType.Method)]
+        [DataTestMethod]
+        public void InitializeShouldValidateTestRunParameter(string arg, string runSettingsWithTestRunParameters)
+        {
+            var args = new string[] { arg };
+
+            this.executor.Initialize(args);
+
+            Assert.IsNotNull(this.settingsProvider.ActiveRunSettings);
+            Assert.AreEqual(runSettingsWithTestRunParameters, settingsProvider.ActiveRunSettings.SettingsXml);
+        }
+
+        [DynamicData(nameof(TestRunParameterArgInvalidTestCases), DynamicDataSourceType.Method)]
+        [DataTestMethod]
+        public void InitializeShouldThrowErrorIfTestRunParameterNodeIsInValid(string arg)
+        {
+            var args = new string[] { arg };
+            var str = string.Format(CommandLineResources.InvalidTestRunParameterArgument, arg);
+
+            CommandLineException ex = Assert.ThrowsException<CommandLineException>(() => this.executor.Initialize(args));
+
+            Assert.AreEqual(str, ex.Message);
+        }
+
+        public static IEnumerable<object[]> TestRunParameterArgInvalidTestCases()
+        {
+            return new List<object[]>
+            {
```

Comment:
This will get initialized every time this method is called. Having private method as you had earlier is actually better